### PR TITLE
 Uptading null languages from WORKFLOW_TASK to default language for Upgrade Task

### PR DESCRIPTION
**Pull request by victoralfaro-dotcms**
Before removing any duplicates make sure that null language_id records are updated to the default language.

### CODE REVIEW

File: dotCMS/src/test/java/com/dotmarketing/common/db/DotConnectTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    private DotConnect dotConnect;
+
+    @Before
+    public void setup() {
+        dotConnect = new DotConnect();
+    }
+
+    @Test
+    public void test_addParamIf() {
+        final Long l = 123L;
```
Comment (dev-dotcms) [inline dotCMS/src/test/java/com/dotmarketing/common/db/DotConnectTest.java:22]: Codacy found an issue: [Avoid variables with short names like l](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5822405)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java [hunk 2]
```diff
@@ -6,6 +6,7 @@
 import com.dotmarketing.exception.DotDataException;
 import com.dotmarketing.exception.DotRuntimeException;
 import com.dotmarketing.startup.StartupTask;
+import com.dotmarketing.util.Config;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java:9]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.util.Config'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5822405)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java [hunk 3]
```diff
@@ -40,22 +41,36 @@ public void executeUpgrade() throws DotDataException, DotRuntimeException {
...
         }
+
         final DotConnect dotConnect = new DotConnect();
-        
+
+        // 18524: set to default language records with null language id
+        try {
+            final long defaultLangId = dotConnect.setSQL("select id from language").loadLong("id");
+            dotConnect.executeStatement(String.format(
+                    "update workflow_task set language_id='%s', mod_date=%s where language_id is null",
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java:51]: use `.addParam` instead

Comment (dsilvam) [reply dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java:51]: `dotConnect.setSQL(update_sql).addParam(param1)
.addParam(param2).loadResult();`

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java [hunk 4]
```diff
@@ -40,22 +41,36 @@ public void executeUpgrade() throws DotDataException, DotRuntimeException {
...
+                    defaultLangId,
+                    getDbDate()));
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java:53]: Let's send the date from dotcms side. Use `addParam(new Date())`

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java [hunk 5]
```diff
@@ -80,12 +96,21 @@ public void executeUpgrade() throws DotDataException, DotRuntimeException {
...
-        
+
         try {
             dotConnect.executeStatement(ADD_CONSTRAINT_SQL);
         } catch (SQLException e) {
             throw new DotDataException(e.getMessage(), e);
         }
     }
+
+    private String getDbDate() {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java:107]: Codacy found an issue: [Avoid unused private methods such as 'getDbDate()'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5822405)

File: dotCMS/src/main/java/com/dotmarketing/common/db/DotConnect.java [hunk 6]
```diff
@@ -38,7 +38,7 @@
 
 /**
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/common/db/DotConnect.java:39]: Please ignore the changes to this file.

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java (hunk 3 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java]: Done.

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java (hunk 4 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java]: Using Date() now.

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java (hunk 5 above)
Comment (erickgonzalez) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java:107]: If this is not used, delete it.

**Review by erickgonzalez — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java (hunk 5 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java]: Yes, is it deleted now.

**Review by dsilvam — APPROVED** [review]

**Review by freddyucv — APPROVED** [review]

**Pull request merged**